pkg/generate/app/app.go

```diff
@@ -431,8 +432,13 @@ func (r *DeploymentConfigRef) DeploymentConfig() (*deployapi.DeploymentConfig, e
 		r.Name = name
 	}
 
-	selector := map[string]string{
-		"deploymentconfig": r.Name,
+	var selector map[string]string
+	if len(r.Labels) > 0 {
```

Comment:
Do we support selector inheritance as I described?  Inherit the labels by default?

I think you're still going to need to add a separate label for the deployment config name, otherwise two deployment configs created from the same new-app would have the same selector. which is bad.